### PR TITLE
[luci/svc] Revise BatchMatMul shape inference

### PR DESCRIPTION
**Pull request by seanshpark**
This will revise BatchMatMul shape inference to pass and show log for shape condition.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #12060

File: compiler/luci/service/src/CircleShapeInferenceRule.cpp [hunk 1]
```diff
@@ -483,7 +483,10 @@ loco::NodeShape infer_batchmatmul_shape(const loco::TensorShape &x_shape,
   loco::Dimension y_rhs = adj_y ? y_shape.dim(y_rank - 2) : y_shape.dim(y_rank - 1);
 
   if (x_rhs.known() && y_lhs.known() && not(x_rhs == y_lhs))
```
Comment (jinevening) [inline compiler/luci/service/src/CircleShapeInferenceRule.cpp:485]: Is it okay for `x_rhs != y_lhs` ? TFLite interpreter (and circle interpreter) does not allows such a shape.

Comment (seanshpark) [reply compiler/luci/service/src/CircleShapeInferenceRule.cpp:485]: I think the cause is from circle-mlir which seems fix is needed.
Thanks for taking a look at tflite interpreter codes :)

**Comment by seanshpark** [comment]
closing as strange shape was from circle-mir converter...

**Pull request closed without merging**